File: Listas/Lista Não Ordenadas/Encadeada (Simples)/Base/Demonstracao_Lista_N_Ordenada_Base.c

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "Lista_N_Ordenada_Base.h"

int main(){
    
    printf("Iniciando Demonstração!\n");
    Lista lista;
    iniciaLista(&lista);
    imprimeLista(&lista);
    inserirLista(&lista,33);
    inserirLista(&lista,10);
    inserirLista(&lista,7);
    inserirLista(&lista,15);
    inserirLista(&lista,14);
    inserirLista(&lista,3);
    inserirLista(&lista,4);
    imprimeLista(&lista);

    printf("Demonstrando Exibir:\n");
    
    if (verificaVazia(&lista)){
        printf("Está Vazia\n");
    }

    printf("Inicio: %d\n",inicioLista(&lista));
    printf("Final: %d\n",finalLista(&lista));
    printf("Tamanho: %d\n",tamanhoLista(&lista));

    printf("\n\nDemonstrando Remoção:");
    printf("Remove o inicio...\n");
    removeListaPrimeiro(&lista);
    imprimeLista(&lista);
    printf("Remove o final...\n");
    removeListaUltimo(&lista);
    imprimeLista(&lista);
    printf("Remove o 15...\n");
    removeListaEspecifico(&lista,15);
    imprimeLista(&lista);
    printf("Tamanho: %d\n",tamanhoLista(&lista));

    printf("\nDemonstração Zerar Lista:");
    printf("Esvazie a Lista...");
    destroiLista(&lista);
    if (verificaVazia(&lista)){
        printf("Está Vazia\n");
    }
    imprimeLista(&lista);

    printf("\nFinalizando...");
    return 0;
}

File: Listas/Lista Não Ordenadas/Encadeada (Simples)/Base/Demonstracao_Lista_N_Ordenada_Base.py
from Lista_N_Ordenada_Base import *

print('Iniciando Demonstração!\n')
lista = List()
lista.inserirLista(33)
lista.inserirLista(10)
lista.inserirLista(7)
lista.inserirLista(15)
lista.inserirLista(14)
lista.inserirLista(3)
lista.inserirLista(4)

print('Demonstrando Exibir:')
print('Lista Completa: ',lista)
if lista.verificaVazia():
    print('Está Vazia')
print('Inicio: ',lista.inicioLista())
print('Final: ',lista.finalLista())
print('Tamanho: ',lista.tamanhoLista())

print('\nDemonstrando Remoção:')
print('Remove o inicio...')
lista.removeListaPrimeiro()
print('Lista Completa: ',lista)
print('Remove o final...')
lista.removeListaUltimo()
print('Lista Completa: ',lista)
print('Remove o 15...')
lista.removeListaEspecifico(15)
print('Lista Completa: ',lista)
print('Tamanho: ',lista.tamanhoLista())

print('\nDemonstração Zerar Lista:')
print('Esvazie a Lista...')
lista.zerarLista()
if lista.verificaVazia():
    print('A lista está Vazia')

print('Lista Completa: ',lista)

print('\nFinalizando...')